protocol/messaging/entrypoints.md

```diff
@@ -0,0 +1,195 @@
+## Summary
+
+This design document introduces `Entrypoint` contracts as a new primitive that allows anyone to add custom logic on top of the `L2ToL2CrossDomainMessenger`.
+It generalizes the `L2ToL2CrossDomainMessenger` design and unlocks other interop primitives such as message batching and expiring.
+To do so, the `L2ToL2CrossDomainMessenger` will allow to authorize a single address to relay a message. The authorized address can then add custom logic before executing the call.
```

Comment:
From our call: let's clarify the text a bit to make it more clear this is a diff to the L2ToL2CrossDomainMessenger that lets you require the call be relayed by the `entrypoint` address

Reply:
Just made a large refactor to the design doc. It should be more clear now, let me know.